### ISSUE TITLE
[indent] Incorrect indentation error inside constructors

### ISSUE THREAD

**Issue by eestein** [typescript-eslint/typescript-eslint#2614]
- [x] I have tried restarting my IDE and the issue persists.
- [x] I have updated to the latest version of the packages.
- [x] I have [read the FAQ](https://github.com/typescript-eslint/typescript-eslint/blob/master/docs/getting-started/linting/FAQ.md) and my problem is not listed.

**Repro**

I noticed the indentation inside constructors is throwing incorrect errors. For instance, the following, valid code:

```ts
constructor(
    @Inject(DOCUMENT) document: any
) {
    this.document = document;
}
```

throws the error:
![Screen Shot 2020-09-01 at 14 51 35](https://user-images.githubusercontent.com/2049255/91888595-f59eda00-ec62-11ea-9b75-301f86005c8b.jpg)

I noticed it only happens if the decorator property is the first (or only one) listed, if you add another one like:

```ts
constructor(
    x: string,
    @Inject(DOCUMENT) document: any
) {
    this.document = document;
}
```

or if you add a modifier, like:

```ts
constructor(
    @Inject(DOCUMENT) private _document: any
) {
    this.document = _document;
}
```

the error goes away.

The "funny" thing is that even comments to disable the rule are affected, so the only solution for now seems to be actually adding a modifier...

![Screen Shot 2020-09-01 at 15 02 26](https://user-images.githubusercontent.com/2049255/91889487-43681200-ec64-11ea-9b91-0742c136e92b.jpg)

**Expected Result**

No error should be displayed

**Actual Result**

Errors are displayed

**Versions**

| package                            | version |
| ---------------------------------- | ------- |
| `@typescript-eslint/eslint-plugin` | `4.3.0` |
| `@typescript-eslint/parser`        | `4.3.0` |
| `TypeScript`                       | `4.0.3` |
| `ESLint`                           | `7.10.0` |
| `node`                             | `12.1.0` |

**Comment by yeonjuan** [comment]
Duplicate of #1232 which merged with issue #1824.

**Comment by eestein** [comment]
Thanks. As recommended on the issue, I've turned off the `no-shadow` and changed to use the plugin one.
